### DISCUSSION TITLE
a smarter way of handling repos of animal GEMs

### DISCUSSION THREAD

**Discussion by haowang-bioinfo** [SysBioChalmers/Human-GEM#386] [Ideas]
This discussion was triggered by question of @johan-gson posted [here](https://github.com/SysBioChalmers/Zebrafish-GEM/pull/6):

- A number of GEM-type repos of animal species have been created and updated regularly using Human-GEM as template
- Would be good to reduce the manual work in curation and reviews, etc
- Suggestion ideas are welcome

**Comment by johan-gson** [comment]
Yeah, I just saw that there is a lot of manual work, reviews etc. for each species. It would probably be valuable to reduce that at some point.

**Reply by mihai-sysbio** [reply]
This was originally discussed when these repositories were instantiated. There are most likely pros and cons in either approaches - mono-repo vs separate repositories.

**Comment by johan-gson** [comment]
Perhaps reviews are only necessary for the first - the others are likely very similar?

**Comment by johan-gson** [comment]
But maybe that requires som automatic test cases or similar, I don't know if any such exist.

**Reply by mihai-sysbio** [reply]
It's envisioned that some automatic validation would be done via `standard-GEM`, but there is no plan yet for "test cases".

**Reply by haowang-bioinfo** [reply]
The design of template-based approach **IS** to reduce manual work in maintenance and curation from the very beginning.

To further reduce manual intervention and increase automation, would suggest:
- Continue to centralize routine functions into `Human-GEM`, on which animal GEM repos become `thin clients` and more dependant
- Develop GitHub actions to `replace` review works whenever possible

**Reply by haowang-bioinfo** [reply]
A generic and centralized animalGEM releasing function would probably be nice to include in addressing this and other related issues ([#7](https://github.com/SysBioChalmers/Fruitfly-GEM/issues/7) and [#970](https://github.com/MetabolicAtlas/MetabolicAtlas/issues/970)). It could also evolve into GH actions run by bot later.

**Reply by mihai-sysbio** [reply]
I would propose an even more generic function called `useAsTemplate(inputGEM)`.
